### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by AShozi**
added more unit tests to increase coverage to 52%

### CODE REVIEW

File: phantomgames/Home/HomeScreenRepository.swift [hunk 1]
```diff
@@ -9,10 +9,12 @@ typealias HomeScreenResult = (Result<[Game], APIError>) -> Void
 typealias GameDetailResult = (Result<GameDetail, APIError>) -> Void
 
 protocol HomeScreenRepositoryType: AnyObject {
+    
     func fetchAPIImageCollectionView(completion: @escaping(HomeScreenResult))
     func fetchAPIImageTableView(completion: @escaping(HomeScreenResult))
     func fetchHomeResultsForCollectionView(completion: @escaping (HomeScreenResult))
     func fetchHomeResultsForTableView(completion: @escaping (HomeScreenResult))
+
```
Comment (ksookdhew) [inline phantomgames/Home/HomeScreenRepository.swift]: Was this empty line on purpose?

File: phantomgamesTests/GameDetailViewModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        }
+        
+        func show(error: String) {
+            showErrorCalled = true
+            errorMessage = error
+        }
+    }
+    
+    func testAddToFavorites() {
+
```
Comment (ksookdhew) [inline phantomgamesTests/GameDetailViewModelTests.swift]: For consistency I would remove this empty line

File: phantomgamesTests/HomeScreenRepositoryTests.swift [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            case .success(let games):
+                XCTAssertFalse(games.isEmpty, "Expected non-empty game list")
+            case .failure(let error):
+                XCTFail("Expected success but got failure: \(error)")
+            }
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 5, handler: nil)
+    }
+    func testFetchGameDetailResultsSuccess() {
```
Comment (ksookdhew) [inline phantomgamesTests/HomeScreenRepositoryTests.swift:81]: add a line

File: phantomgamesTests/UIViewControllerDisplayAlertTests.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        var presentedAlert: UIAlertController?
+        
+        override func present(_ viewControllerToPresent: UIViewController, animated flag: Bool, completion: (() -> Void)? = nil) {
+            presentCalled = true
+            presentedAlert = viewControllerToPresent as? UIAlertController
+        }
+    }
+    
+    func testDisplayAlertSuccess() {
+
```
Comment (ksookdhew) [inline phantomgamesTests/UIViewControllerDisplayAlertTests.swift]: I would remove empty line

File: phantomgamesTests/URLSessionExtensionTests.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                expectation.fulfill()
+            case .failure(let error):
+                XCTFail("Request should not fail: \(error)")
+            }
+        }
+        waitForExpectations(timeout: 5, handler: nil)
+    }
+    
+    func testRequestFailure() {
+
```
Comment (ksookdhew) [inline phantomgamesTests/URLSessionExtensionTests.swift]: I would remove empty line

File: phantomgames/Home/HomeScreenRepository.swift (hunk 1 above)
Comment (AShozi) [reply phantomgames/Home/HomeScreenRepository.swift]: nice catch thanks

**Review by ksookdhew — APPROVED** [review]
Nice 👏

**Review by kiasha10 — APPROVED** [review]

File: phantomgames/GameDetails/GameDetailViewController.swift [hunk 6]
```diff
@@ -83,6 +87,14 @@ class GameDetailViewController: UIViewController {
             gameButton.isEnabled = false
         }
     }
+    private func updateFavoriteButton() {
+        if gameDetailViewModel.isFavorite {
```
Comment (sergembamba) [inline phantomgames/GameDetails/GameDetailViewController.swift:91]: let image =  gameDetailViewModel.isFavorite  ? UIImage(systemName: "star.fill") : UIImage(systemName: "star"),
 favoriteButton.setImage(image, for: .normal)

File: phantomgames/GameDetails/GameDetailViewModel.swift [hunk 7]
```diff
@@ -18,11 +18,13 @@ class GameDetailViewModel {
     private let repository: GameDetailRepositoryType
     private var gameID = 0
     private var gameDetail: GameDetail?
+    private var coreDataManager: CoreDataModel
```
Comment (sergembamba) [inline phantomgames/GameDetails/GameDetailViewModel.swift:21]: Maybe let?

**Pull request merged**